src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -296,7 +293,7 @@ internal BoundExpression ConvertPatternExpression(TypeSymbol inputType, CSharpSy
                 hasErrors = true;
             }
 
-            DeclareLocalVariable(localSymbol, identifier, declType);
+            localSymbol.SetType(declType);
```

Comment:
I didn't notice that bug in the code before, but now that `DeclareLocalVariable` is inlined, it is easy to see. I will remove this.